src/types.js

```diff
@@ -316,6 +316,14 @@ export type MessageLike =
 //   https://formatjs.io/docs/react-intl/api/#formatmessage
 type MessageFormatPrimitiveValue = string | number | boolean | null | void;
 
+// For rich text formatting:
+//   https://formatjs.io/docs/react-intl/components/#rich-text-formatting
+// Name found in the docs:
+//   https://formatjs.io/docs/react-intl/api/#formatmessage
+// Type taken by hand from TypeScript.
+// eslint-disable-next-line ft-flow/type-id-match
+type FormatXMLElementFn = (parts: $ReadOnlyArray<string | Node>) => Node;
```

Comment:
nit: `FormatXmlElementFn`, for proper camel case as defined here: https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined

The rationale is that that makes it unambiguous where the word boundaries are, even when some of the "words" are all-caps acronyms.

(I feel like I've mentioned this somewhere before, either to you or someone else. Should really put a mention of it in our style guide.)

Reply:
Like the type just above this one—

```js
// Name and type copied from docs:
//   https://formatjs.io/docs/react-intl/api/#formatmessage
type MessageFormatPrimitiveValue = string | number | boolean | null | void;
```

—the name was copied from upstream. I'll change the name to be camel case, but leave the upstream name in a comment so the reader can find the upstream definition more easily.

Reply:
Ah, I think I'd missed that the name came from upstream's docs. Still, it seems like it's *only* in the docs (and not very prominent there either -- I don't even see it actually defined there, let alone e.g. having its own entry), not an API we're interacting with, so happy to have the name adapted to our style.